src/nvim/terminal.c

```diff
@@ -331,7 +331,10 @@ void terminal_close(Terminal *term, char *msg)
   }
 
   if (buf) {
+    aco_save_T aco;
+    aucmd_prepbuf(&aco, buf);
     apply_autocmds(EVENT_TERMCLOSE, NULL, NULL, false, buf);
+    aucmd_restbuf(&aco);
```

Comment:
I think the pattern would be to use `bufref_valid` instead, as with https://github.com/vim/vim/blob/b4d9b893d3c7af24b5adf4ae0475ada32d36f5ad/src/buffer.c#L563-L573 (NOTE: requires missing Vim patches to get to this)